modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -438,34 +469,104 @@ else if (Objects.equals(parameterName, "relatedObjectEntryId")) {
 		return parameters;
 	}
 
+	private ObjectDefinition _getRelatedObjectDefinition(
+		ObjectRelationship objectRelationship) {
+
+		if (_objectDefinition.getObjectDefinitionId() ==
+				objectRelationship.getObjectDefinitionId2()) {
+
+			return _objectDefinitionLocalService.fetchObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+		}
+
+		return _objectDefinitionLocalService.fetchObjectDefinition(
+			objectRelationship.getObjectDefinitionId2());
+	}
+
 	private Map<ObjectRelationship, ObjectDefinition>
-		_getRelatedObjectDefinitionsMap() {
+			_getRelatedObjectDefinitionsMap()
+		throws Exception {
 
 		Map<ObjectRelationship, ObjectDefinition> relatedObjectDefinitionsMap =
 			new HashMap<>();
 
 		List<ObjectRelationship> objectRelationships =
-			_objectRelationshipLocalService.getObjectRelationships(
+			_objectRelationshipLocalService.getAllObjectRelationships(
 				_objectDefinition.getObjectDefinitionId());
 
 		for (ObjectRelationship objectRelationship : objectRelationships) {
-			ObjectDefinition objectDefinition =
-				_objectDefinitionLocalService.fetchObjectDefinition(
-					objectRelationship.getObjectDefinitionId2());
-
 			relatedObjectDefinitionsMap.put(
-				objectRelationship, objectDefinition);
+				objectRelationship,
+				_getRelatedObjectDefinition(objectRelationship));
 		}
 
 		return relatedObjectDefinitionsMap;
 	}
 
+	private Schema _getSchema(
+		ObjectRelationship objectRelationship, String schemaName) {
+
+		ObjectSchema schema = new ObjectSchema();
+
+		String description = StringBundler.concat(
+			"Information about the relationship ", objectRelationship.getName(),
+			" can be embedded with \"nestedFields\".");
+
+		schema.setDescription(description);
+
+		schema.set$ref(schemaName);
+
+		if (((objectRelationship.getObjectDefinitionId1() ==
+				_objectDefinition.getObjectDefinitionId()) ||
+			 Objects.equals(
+				 objectRelationship.getType(),
+				 ObjectRelationshipConstants.TYPE_MANY_TO_MANY)) &&
+			(schemaName != null)) {
+
+			ArraySchema arraySchema = new ArraySchema();
+
+			arraySchema.setItems(schema);
+
+			arraySchema.setDescription(description);
+
+			return arraySchema;
+		}
+
+		return schema;
+	}
+
+	private void _setSchemaDescription(
```

Comment:
What is the purpose of this method?

I have been debugging and it includes a description in the schema, but if there is more than one relationship, it is overriden with the latest execution, makes no sense

![image](https://user-images.githubusercontent.com/32699538/219598819-1a7b0dbf-d3b4-4635-8a44-824ee1ad5ca7.png)

Reply:
@ccorreagg the purpose of this method is to set the description to the schema that is related to a One-to-Many relationship. Because in the example of `University oneToMany Students`: if you request the Schema of the Student, the field `universityStudent` doesn't have any description because OpenAPI 3.0.1 doesn't support this. So I told this Pablo and we came to the conclusion that there should be a description and must be in the Shema of the related element.

Reply:
Ok, so, maybe it should be interesting including it in the PR description or in the commit message, with a relation to the thread were that was discussed